File: src/content/docs/guides/fact/fact.go
package main

import (
	"fmt"
	"log"
)

func factorial(n int) int {
	if n == 0 || n == 1 {
		return 1
	}
	return n * factorial(n-1)
}

func main() {

	fmt.Println("enter a number:")
	var number int
	_, err := fmt.Scanln(&number)
	if err != nil {
		log.Fatal("Invalid input. Please enter a valid non-negative integer.")
	}

	if number > 20 {
		log.Fatal("cannot compute more than fact(20) overflow")
	}
	if number < 0 {
		log.Fatal("Invalid input. Please enter a non-negative integer.")
	}

	fmt.Println(factorial(number))
}

File: src/content/docs/guides/fact/Fact.java
import java.util.Scanner;

public class Fact {
    public static void main(String[] args) {
        System.out.println("enter a number:");
        Scanner input = new Scanner(System.in);
        try {
            int n = Integer.parseInt(input.nextLine());
            if (n < 0){
                System.out.println("Invalid input. Please enter a non-negative integer.");
                return;
            }

            System.out.println(factorial(n));
        } catch (NumberFormatException nfe) {
            System.out.println("Invalid input. Please enter a valid non-negative integer.");
        } finally {
            input.close();
        }

    }

    public static int factorial(int n) {
        int result = 1;
        for (int i = 1; i <= n; i++) {
          result = result * i;
        }
        return result;
    }
}